testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java

```diff
@@ -95,12 +115,46 @@ public void testRuntime(TestInfo testInfo, List<CodeQuarkusExtensions> extension
             LOGGER.info("Removing repositories and pluginRepositories from pom.xml ...");
             removeRepositoriesAndPluginRepositories(appDir + File.separator + "pom.xml");
             adjustPrettyPrintForJsonLogging(appDir.getAbsolutePath());
+            
+            List<String> cmd;
+            // Build
+            if (mvnCmds != MvnCmds.MVNW_DEV) {
+                buildLogA = new File(logsDir + File.separator + "build.log");
+                ExecutorService buildService = Executors.newFixedThreadPool(1);
+                List<String> baseBuildCmd = new ArrayList<>();
+                baseBuildCmd.addAll(Arrays.asList(mvnCmds.mvnCmds[0]));
+                cmd = getBuildCommand(baseBuildCmd.toArray(new String[0]));
+
+                appendln(whatIDidReport, "# " + cn + ", " + mn);
+                appendln(whatIDidReport, (new Date()).toString());
+                appendln(whatIDidReport, appDir.getAbsolutePath());
+                appendln(whatIDidReport, "Extensions: " + extensions.toString());
+                appendlnSection(whatIDidReport, String.join(" ", cmd));
+
+                LOGGER.info("Building (" + cmd + ")");
+                buildService.submit(new Commands.ProcessRunner(appDir, buildLogA, cmd, 20));
+
+                buildService.shutdown();
+                buildService.awaitTermination(30, TimeUnit.MINUTES);
+
+                assertTrue(buildLogA.exists());
+            }
+            
+            // Run
             runLogA = new File(logsDir + File.separator + "dev-run.log");
-            LOGGER.info("Running command: " + devCmd + " in directory: " + appDir);
+            if (mvnCmds == MvnCmds.MVNW_DEV) {
+                cmd = getBuildCommand(mvnCmds.mvnCmds[0]);
+            } else {
+                cmd = getRunCommand(mvnCmds.mvnCmds[1]);
+            }
+            
+            LOGGER.info("Running (" + cmd + ") in directory: " + appDir);
+            pA = runCommand(cmd, appDir, runLogA);
+
             appendln(whatIDidReport, "Extensions: " + extensions.toString());
             appendln(whatIDidReport, appDir.getAbsolutePath());
-            appendlnSection(whatIDidReport, String.join(" ", devCmd));
-            pA = runCommand(devCmd, appDir, runLogA);
+            appendlnSection(whatIDidReport, String.join(" ", cmd));
+            
```

Comment:
`pA = runCommand(cmd, appDir, runLogA);` from line 152 should be here, after `whatIDidReport` details are saved.

Reply:
Changed. In `SpecialCharsTest.java`, the `appendlnSection` methods are also done after the `runCommand` statement, should I change it here as well?

Reply:
Yes, dedicated PR please.

Reply:
https://github.com/quarkus-qe/quarkus-startstop/pull/58